src/jit/codegen.h

```diff
@@ -548,13 +548,13 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     //-------------------------------------------------------------------------
     // scope info for the variables
 
-    void genSetScopeInfo(unsigned            which,
-                         UNATIVE_OFFSET      startOffs,
-                         UNATIVE_OFFSET      length,
-                         unsigned            varNum,
-                         unsigned            LVnum,
-                         bool                avail,
-                         Compiler::siVarLoc& loc);
+    void genSetScopeInfo(unsigned       which,
+                         UNATIVE_OFFSET startOffs,
+                         UNATIVE_OFFSET length,
+                         unsigned       varNum,
+                         unsigned       LVnum,
+                         bool           avail,
+                         siVarLoc*      loc);
```

Comment:
I think @sandreenko 's suggestion was to make this `const siVarLoc& loc` rather than `siVarLoc* loc`.

Reply:
Oh sorry, it can't be a "const siVarLoc*" due to the fact that it is possible modified for x86 machines https://github.com/dotnet/coreclr/blob/master/src/jit/codegencommon.cpp#L10896

I can encapsulate that part of code as a method and called it before genSetScopeInfo so as to be a const * at least in genSetScopeInfo.

Reply:
Ah, x86, didn't notice that ifdef :)

I'm a bit baffled by this code. There's a bunch of code which attempts to build a `siVarLoc` only to have another bunch of code immediately modify it. Oh well, I've seen stranger things in the JIT.

I'm not sure if it's worth modifying this. Especially considering the "return" in the middle of that code.

Reply:
I think this is quite reasonable to leave as a pointer. I might have changed the name to `pLoc` to make that clear, and perhaps a comment in the header about why this can't be `const` would be good.

Reply:
The purpose of this PR is to:

1) Construct a siVarLoc from outside this code if its required.
2) Make the genSetScopeInfo() simpler to read (much of that code was arquirecture dependent.

It's fine for me if you want me to start encapsulating code if it makes it more cleanear. I don't know if it is a good idea to include all in one PR.

Reply:
> I don't know if it is a good idea to include all in one PR.

Probably not. At 500 lines is already rather large (says the person having 1000 lines PRs :grin:), even if it mostly just moves code around.

Reply:
> I think this is quite reasonable to leave as a pointer. I might have changed the name to `pLoc` to make that clear, and perhaps a comment in the header about why this can't be `const` would be good.

I have renamed it as varLoc just to be consistent with the implementation, documentation and some other uses of that type